### DISCUSSION TITLE
attribute naming conventions

### DISCUSSION THREAD

**Discussion by timker** [lit/lit#2179] [General]
Is their a preference for hyphens attribute properties?

@property({ attribute:"video-id" }) videoId: string;
vs
@property() videoId: string;

Hyphens seem more correct for attributes.
camelCase seems more correct for properties
however does it make it unnecessarily complex when both are so entwined with each other?

**Comment by jhorback** [comment]
I use hyphens for attributes and camelCase for properties as you mentioned. 

I don't find it unnecessarily complex since you access attributes and properties quite differently.

**Comment by KeithHenry** [comment]
I use camelCase for properties and hyphen-lower-case for attributes, I find it helps be clear about which I'm setting:

```ts
@Property({ attribute:"video-id" }) 
declare videoId: string;

@Property({ attribute:false }) 
declare videoOptions: VideoOptionsInterface;
```

Then it's clear in my usage:

```
const vid = '...';
const opts: VideoOptionsInterface = { ... };
return html`
  <my-video 
    video-id=${vid} 
    .videoOptions=${opts}></my-video>`
```

**Comment by busynest** [comment]
a custom property is "data-"

**Reply by timker** [reply]
technically they are "data attributes"

and they _kinda_ follow the kebab => camelcase mapping


e.g 
```html
<div data-some-thing=5 ></div>
```
would be 
```js
div.dataset.someThing  === "5"
```

so a weird **data** attribute => **dataset** property mapping, but then it's consistent
